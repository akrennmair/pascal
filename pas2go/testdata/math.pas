program test;

var r : real;
	i : integer;
	c : char;
	b : boolean;

begin
	r := cos(23.5);
	writeln('cos(23.5) = ', r);
	r := frac(23.5);
	writeln('frac(23.5) = ', r);
	r := int(23.5);
	writeln('int(23.5) = ', r);
	r := ln(23.5);
	writeln('ln(23.5) = ', r);
	r := pi;
	writeln('pi = ', r);
	r := sin(23.5);
	writeln('sin(23.5) = ', r);
	r := sqr(23.5);
	writeln('sqr(23.5) = ', r);
	r := sqrt(23.5);
	writeln('sqrt(23.5) = ', r);
	i := trunc(23.6);
	writeln('trunc(23.6) = ', i);
	i := round(23.6);
	writeln('round(23.6) = ', i);
	c := chr(40);
	writeln('chr(40) = ', c);
	b := odd(2);
	writeln('odd(2) = ', b);
	b := odd(23);
	writeln('odd(23) = ', b)
end.
