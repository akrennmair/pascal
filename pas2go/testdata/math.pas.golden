package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program test
func main() {
	var (
		r float64
		i int
		c byte
		b bool
	)
	_ = r
	_ = i
	_ = c
	_ = b

	r = system.Cos(23.5e0)
	system.Writeln("cos(23.5) = ", r)
	r = system.Frac(23.5e0)
	system.Writeln("frac(23.5) = ", r)
	r = system.Int(23.5e0)
	system.Writeln("int(23.5) = ", r)
	r = system.Exp(23.5e0)
	system.Writeln("ln(23.5) = ", r)
	r = system.Pi()
	system.Writeln("pi = ", r)
	r = system.Sin(23.5e0)
	system.Writeln("sin(23.5) = ", r)
	r = system.Sqr(23.5e0)
	system.Writeln("sqr(23.5) = ", r)
	r = system.Sqrt(23.5e0)
	system.Writeln("sqrt(23.5) = ", r)
	i = system.Trunc(23.6e0)
	system.Writeln("trunc(23.6) = ", i)
	i = system.Round(23.6e0)
	system.Writeln("round(23.6) = ", i)
	c = system.Chr(40)
	system.Writeln("chr(40) = ", c)
	b = system.Odd(2)
	system.Writeln("odd(2) = ", b)
	b = system.Odd(23)
	system.Writeln("odd(23) = ", b)
}
